assets/prototype/domain/eventEditor/datetimes/dateCard/EditDateButton.tsx

```diff
@@ -2,10 +2,18 @@ import React, { CSSProperties } from 'react';
 import { Button } from '@blueprintjs/core/lib/esm';
 import { EditItemButtonProps } from '../../types';
 import { useDatetimeContext } from './../../hooks';
+import { useEditorModal } from '../../../../application/ui/components/layout/eeditorModal';
 
 const EditDateButton: React.FC<EditItemButtonProps> = ({ position }): JSX.Element => {
-	const { editors, editorState } = useDatetimeContext();
-	const onClick = (): void => editorState.openEditor(editors.editForm);
+	const { id: entityId } = useDatetimeContext();
+
+	const { openEditor } = useEditorModal();
+
+	const onClick = (): void =>
+		openEditor({
+			editorId: 'editDatetime',
```

Comment:
can this use the same kind of idea as the `editors` var in `assets\prototype\domain\eventEditor\context\DateTimeProvider` ?

So this would be something like:

```js
openEditor({
	editorId: editors.editForm,
	entityId,
});
```

Reply:
That's not needed as I have used a different approach where each item of the modal stack will be an object where you can pass any data you want:

```ts
interface EditorState {
	editorId: EditorId;
	isOpen: boolean;
	entityId?: EntityId;
	[key: string]: any;
}

type EditorStack = EditorState[];
```